src/org/thoughtcrime/securesms/ShowUserDetailsActivity.java

```diff
@@ -0,0 +1,215 @@
+import android.widget.Button;
+import android.widget.TextView;
+
+import org.thoughtcrime.securesms.components.AvatarImageView;
+import org.thoughtcrime.securesms.conversation.ConversationActivity;
+import org.thoughtcrime.securesms.database.Address;
+import org.thoughtcrime.securesms.database.DatabaseFactory;
+import org.thoughtcrime.securesms.database.ThreadDatabase;
+import org.thoughtcrime.securesms.mms.GlideApp;
+import org.thoughtcrime.securesms.mms.GlideRequests;
+import org.thoughtcrime.securesms.recipients.Recipient;
+import org.thoughtcrime.securesms.recipients.RecipientModifiedListener;
+import org.thoughtcrime.securesms.util.DynamicLanguage;
+import org.thoughtcrime.securesms.util.DynamicTheme;
+import org.thoughtcrime.securesms.util.Util;
+import org.thoughtcrime.securesms.util.ViewUtil;
+
+import network.loki.messenger.R;
+
+@SuppressLint("StaticFieldLeak")
+public class ShowUserDetailsActivity extends PassphraseRequiredActionBarActivity implements RecipientModifiedListener
+{
+  private static final String TAG = ShowUserDetailsActivity.class.getSimpleName();
+
+  public static final String ADDRESS_EXTRA           = "recipient_address";
+  public static final String THREAD_ID_EXTRA         = "thread_id";
+  public static final String DISTRIBUTION_TYPE_EXTRA = "distribution_type";
+
+
+  private final DynamicTheme    dynamicTheme    = new DynamicTheme();
+  private final DynamicLanguage dynamicLanguage = new DynamicLanguage();
+
+  private Address                 address;
+
+
+
+  @Override
+  public void onPreCreate() {
+    dynamicTheme.onCreate(this);
+    dynamicLanguage.onCreate(this);
+  }
+
+  @Override
+  public void onCreate(Bundle instanceState, boolean ready) {
+    setContentView(R.layout.show_user_details_activity);
+    this.address = getIntent().getParcelableExtra(ADDRESS_EXTRA);
+
+    Recipient recipient = Recipient.from(this, address, true);
+
+    getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+    setHeader(recipient);
+    recipient.addListener(this);
+    Bundle bundle = new Bundle();
+    bundle.putParcelable(ADDRESS_EXTRA, address);
+    initFragment(R.id.user_details_fragment, new UserDetailsFragment(), null, bundle);
+  }
+
+
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    dynamicTheme.onResume(this);
+    dynamicLanguage.onResume(this);
+  }
+
+
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    switch (item.getItemId()) {
+      case android.R.id.home:{
+        finish();
+        return true;
+      }
+      default:
+        break;
+    }
+    return super.onOptionsItemSelected(item);
+  }
+
+
+  private void setHeader(@NonNull Recipient recipient) {
+    if      (recipient.isGroupRecipient())           setGroupRecipientTitle(recipient);
+    else if (recipient.isLocalNumber())              setSelfTitle();
+    else {
+      String displayName = DatabaseFactory.getLokiUserDatabase(this).getDisplayName(address.serialize());
```

Comment:
`getSupportActionBar().setTitle(recipient.toShortString());`
should do the same thing :)